scripts/linux/nested-edge-deploy-agent.sh

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env bash
+# This script is intended to be used for nested edge tests. It deploy level 4 and level 5.
+# Level 3 is deveployed by specialized script for connectivity/long haul and quick start.
+
+function create_certificates() {
+    echo "Installing test root certificate bundle."    
+
+    echo "Generating edge device certificate"
+    device_name=$(az vm show -d  -g "iotedge-deploy" -n $(hostname) --query fqdns)
+    eval device_name=${device_name}
+    echo "  Hostname FQDN: ${device_name}" 
+
+    /certs/certGen.sh create_edge_device_certificate ${device_name}
+}
+#@TODO this might not be compatible for CENTOS
+function install_and_setup_iotedge() {
+    echo "Install and setup iotedge"
+
+    echo "  Install artifacts"
+    declare -a pkg_list=( $iotedged_artifact_folder/*.deb )
+    iotedge_package="${pkg_list[*]}" 
+    sudo dpkg -i --force-confnew ${iotedge_package}
+ 
+    echo "  Updating IoT Edge configuration file to use the newly installed certificcates"
+    device_ca_cert_path="/certs/certs/iot-edge-device-${device_name}-full-chain.cert.pem"
+    device_ca_pk_path="/certs/private/iot-edge-device-${device_name}.key.pem"
+    trusted_ca_certs_path="/certs/certs/azure-iot-test-only.root.ca.cert.pem"
+    sudo sed -i "165s|.*|certificates:|" /etc/iotedge/config.yaml
+    sudo sed -i "166s|.*|  device_ca_cert: \"$device_ca_cert_path\"|" /etc/iotedge/config.yaml
+    sudo sed -i "167s|.*|  device_ca_pk: \"$device_ca_pk_path\"|" /etc/iotedge/config.yaml
+    sudo sed -i "168s|.*|  trusted_ca_certs: \"$trusted_ca_certs_path\"|" /etc/iotedge/config.yaml
+
+    echo "Updating the device connection string"
+    sudo sed -i "s#\(device_connection_string: \).*#\1\"${CONNECTION_STRING}\"#g" /etc/iotedge/config.yaml
+
+    echo "Updating the device and parent hostname"
+    sudo sed -i "224s/.*/hostname: \"$device_name\"/" /etc/iotedge/config.yaml
+
+    if [ ! -z $PARENT_NAME ]; then
+        echo "Updating the parent hostname"
+        sudo sed -i "237s/.*/parent_hostname: \"$PARENT_NAME\"/" /etc/iotedge/config.yaml
+    fi
+
+    echo "Updating edge Agent"
+    sudo sed -i "207s|.*|    image: \"${CUSTOM_EDGE_AGENT_IMAGE}\"|" /etc/iotedge/config.yaml
+
+    if [ -z $PARENT_NAME ]; then
+        sudo sed -i "208s|.*|    auth:|" /etc/iotedge/config.yaml
+        sed -i "209i\      serveraddress: \"${CONTAINER_REGISTRY}\"" /etc/iotedge/config.yaml
+        sed -i "210i\      username: \"${CONTAINER_REGISTRY_USERNAME}\"" /etc/iotedge/config.yaml
+        sed -i "211i\      password: \"${CONTAINER_REGISTRY_PASSWORD}\"" /etc/iotedge/config.yaml
+    fi
+
+    sudo cat /etc/iotedge/config.yaml
```

Comment:
Does this require sudo?

Reply:
Yes, files in /etc requires sudo access for reading